poly-commitment/src/commitment.rs

```diff
@@ -607,22 +607,22 @@ impl<G: CommitmentCurve> SRS<G> {
         &self,
         domain: D<G::ScalarField>,
         plnm: &Evaluations<G::ScalarField, D<G::ScalarField>>,
-        max: Option<usize>,
     ) -> PolyComm<G> {
-        let is_zero = plnm.evals.par_iter().all(|x| x.is_zero());
-        let basis = match self.lagrange_bases.get(&domain.size()) {
+        let lagrange_basis = self.lagrange_bases.get(&domain.size());
+        let basis = match lagrange_basis.as_ref() {
             None => panic!("lagrange bases for size {} not found", domain.size()),
-            Some(v) => &v[..],
+            Some(v) => v,
+        };
+        let commit_evaluations = |evals: &Vec<G::ScalarField>, basis: &Vec<PolyComm<G>>| {
+            PolyComm::<G>::multi_scalar_mul(&basis.iter().collect::<Vec<_>>()[..], &evals[..])
         };
         match domain.size.cmp(&plnm.domain().size) {
             std::cmp::Ordering::Less => {
                 let s = (plnm.domain().size / domain.size) as usize;
                 let v: Vec<_> = (0..(domain.size())).map(|i| plnm.evals[s * i]).collect();
-                Self::commit_helper(&v[..], basis, None, is_zero, max)
-            }
-            std::cmp::Ordering::Equal => {
-                Self::commit_helper(&plnm.evals[..], basis, None, is_zero, max)
+                commit_evaluations(&v, basis)
             }
+            std::cmp::Ordering::Equal => commit_evaluations(&plnm.evals, basis),
```

Comment:
if we remove `commit_helper` here I think it's only used in a single place, so might as well inline it